packages/features/data-table/lib/types.ts

```diff
@@ -1,6 +1,6 @@
 import { z } from "zod";
```

Comment:
@eunjae-lee heavy refactor of this file to improve TS performance

Reply:
Thanks. I love that you put `satisfies` so that we won't accidentally break it in the future. If `z.infer` causes bad performance, maybe we could have an eslint warning rule?

Reply:
That would warn our entire codebase, z.infer is heavy on TS but it was uniquely heavy here because it was inferring deeply (z.infer<z.infer<z.infer<..)